internal/pkg/inject/offset_results.json

```diff
@@ -2915,6 +3077,168 @@
               }
             ]
           },
+          {
+            "struct": "pattern",
```

Comment:
Similar here. This wasn't added until 1.22.

Reply:
I noticed that if I rebuild the `offsetgen` tool and run it for the first time it gets it right, and the second time it puts all these wrong values, so I'm guessing there is some bug in the cache mechanism there.